cedar-policy-core/src/entities/json/entities.rs

```diff
@@ -229,6 +235,11 @@ impl<'e, 's, S: Schema> EntityJsonParser<'e, 's, S> {
                 }
             })
             .collect::<Result<_, JsonDeserializationError>>()?;
+        // if this `allowed_parent_types` is `None`, then there are no restrictions
+        // (any parent type is allowed)
+        let allowed_parent_types = self
+            .schema
+            .map(|schema| schema.allowed_parent_types(uid.entity_type()));
```

Comment:
I haven't looked at all the surrounding code, so this is an uninformed question, but: why would `allowed_parent_types` be None here, and why does that mean that any parent type should be allowed? The corresponding function in `schema.rs` doesn't return an optional type.

Reply:
It would be `None` if `self.schema` is `None`, i.e., if we're not doing schema-based parsing.  Perhaps this can be clearer somehow?

Reply:
Thinking ahead, it could make sense to return `None` from the impl in `schema.rs` if we wanted to implemented schema based parsing using a partial schema (once that's finished)

Reply:
@khieta I'll update the comment.
@john-h-kastner-aws This is a good point, but I think it makes sense to refactor then / in that PR rather than right now.